assets/src/edit-story/components/inspector/inspectorProvider.js

```diff
@@ -6,26 +6,78 @@ import PropTypes from 'prop-types';
 /**
  * WordPress dependencies
  */
-import { useState } from '@wordpress/element';
+import { useCallback, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import { useAPI } from '../../app/api';
 import Context from './context';
 
 const DESIGN = 'design';
 const DOCUMENT = 'document';
 const PREPUBLISH = 'prepublish';
 
 function InspectorProvider( { children } ) {
+	const { actions: { getAllStatuses, getAllUsers } } = useAPI();
 	const [ tab, setTab ] = useState( DESIGN );
+	const [ users, setUsers ] = useState( [] );
+	const [ statuses, setStatuses ] = useState( [] );
+
+	const [ isUsersLoading, setIsUsersLoading ] = useState( false );
+	const [ isStatusesLoading, setIsStatusesLoading ] = useState( false );
```

Comment:
This ought to be changed to two instances of the same reducer-based hook, that can update a list and remember loading state (and probably handle error too).

Reply:
Did you have something in mind here?

Reply:
This two functions are similar but not the same. The output of the two different api calls are different so require some different logic inside them, making it not easy to refactor into one generate function with params. If I am missing something, please go ahead @barklund give me a code snippet.